code/game/machinery/doors/brigdoors.dm

```diff
@@ -276,71 +277,101 @@
 /obj/machinery/door_timer/attack_hand(mob/user)
 	if(..())
 		return
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/door_timer/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/door_timer/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "brig_timer.tmpl", "Brig Timer", 500, 400)
+		prisoner_suggested_names = list()
+		for(var/mob/living/carbon/human/H in range(4, get_turf(src)))
+			if(H.handcuffed)
+				prisoner_suggested_names += H.name
```

Comment:
Use `tgui_static_data` for this. The `tgui_interact` proc is meant to follow standard format and not have modifications like this in, especially when `tgui_static_data` exists for this very purpose

```dm
/thing/tgui_static_data(mob/user)
	var/list/data = list()
	prisoner_suggested_names = list()
	for(var/mob/living/carbon/human/H in range(4, get_turf(src)))
		if(H.handcuffed)
			prisoner_suggested_names += H.name
	data["prisoner_suggested_names"] = prisoner_suggested_name
	return data
```

You can then interact with this in the UI in the same way as `tgui_data()`